R/util.R

```diff
@@ -69,6 +69,25 @@ read_lines_utf8 <- function(file, encoding) {
   to_utf8(lines, encoding)
 }
 
+# Determine if there are duplicate chunk names in the document's
+# `input_lines`; this allows the render process to terminate early
+# for any long-running rendering tasks
+any_duplicate_chunk_names <- function(input_lines) {
```

Comment:
@yihui Are there any functions or regexs exported from knitr that would facilitate this being done in the most reliable/compatible fashion possible?